core/trino-main/src/main/java/io/trino/operator/join/LookupSource.java

```diff
@@ -30,8 +30,22 @@
 
     long joinPositionWithinPartition(long joinPosition);
 
+    default void getJoinPosition(int[] positions, Page hashChannelsPage, Page allChannelsPage, long[] rawHashes, long[] result)
```

Comment:
nit: javadoc would be nice. It's not always obvious if  `positions.length == rawHashes` or `long rawHash = rawHashes[positions[i]]`